### DISCUSSION TITLE
Ability to serialize to a stream

### DISCUSSION THREAD

**Discussion by CrustyAuklet** [stephenberry/glaze#400] [Ideas]
Presuming that `radio.cfg_` is a structure that I have appropriately configured for glaze with meta information I can make a pretty nice `set` function in my embedded device REPL.

Something like:
```c++
Result<void> RadioTask::ConfigTree::on_set(std::string_view path, std::string_view data) {
    if(glz::read_as_json(radio.cfg_, path, data)) {
        radio->commit();
        return success();
    }
    return Error::BadArgument;
}
```

But a `get` function is a little harder. I don't see a way to write to a stream, so I have to resort to some arbitrarily large static buffer
```c++
Result<void> RadioTask::ConfigTree::on_get(std::string_view path, io::OutDevice& os) {
    // static std::array<char, 128> b;  // couldn't actually get std::array to work!
    static char b[256];
    memset(b, 0, 256);
    if(glz::write_as_json(radio.cfg_, path, b)) {
        os.write(to_bytes(b, strlen(b)));
        return success();
    }
    return Error::BadArgument;
}
```

I tried wrapping our custom `io::OutDevice` class (just a UART virtual interface) with a simple output iterator but glaze concepts want either a "range" (defined by having `begin` and `end`) or a raw buffer.

Let me know if this is out of scope, since I know streams are not the most performant solution. But on small embedded platforms a stream is often the best solution to limit RAM use and stack size.

**Comment by stephenberry** [comment]
Thanks so much for sharing your use case. We initially had streaming support, but it made the code much more complicated. We'll probably add it back at some point, but I don't expect it to be in the near future.

I would recommend sharing your buffer as much as possible so that you don't end up with multiple large buffers in your code. Using static here was a good choice.

I plan to look more at using `std::pmr` with a stack buffer (e.g. `std::pmr::string`) for some of these kinds of use case. You might find this worth looking into. A benefit here is that it could eliminate the need to call `strlen`, which should improve performance.

I'll make an issue about `std::array<char...` not working, since it should.
